dropshot/src/extractor/header.rs

```diff
@@ -71,3 +86,55 @@ where
         get_metadata::<HeaderType>(&ApiEndpointParameterLocation::Header)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use schemars::JsonSchema;
+    use serde::Deserialize;
+
+    use crate::{extractor::header::http_request_load_header, RequestInfo};
+
+    #[test]
+    fn test_header_parsing() {
+        #[allow(dead_code)]
+        #[derive(Debug, Deserialize, JsonSchema)]
+        pub struct TestHeaders {
+            header_a: String,
+            #[serde(rename = "X-Header-B")]
+            header_b: String,
+        }
+
+        let addr = std::net::SocketAddr::new(
+            std::net::Ipv4Addr::LOCALHOST.into(),
+            8080,
+        );
+
+        let request =
+            hyper::Request::builder().uri("http://localhost").body(()).unwrap();
+        let info = RequestInfo::new(&request, addr);
+
+        let parsed = http_request_load_header::<TestHeaders>(&info);
+        assert!(parsed.is_err());
+
+        let request = hyper::Request::builder()
+            .header("header_a", "header_a value")
+            .header("X-Header-B", "header_b value")
```

Comment:
If we used `x-hEaDeR-b` here as the key, would this value still make it to the server?